### ISSUE TITLE
Integration tests are broken after composer update (phpunit 9.5.0)

### ISSUE THREAD

**Issue by fooman** [magento/magento2#33761]
### Preconditions (*)

1. Magento 2.4.3 (but I suspect earlier versions as well due to the phpunit constraint ^9.0

### Steps to reproduce (*)
1. install Magento 2.4.3
2. run composer update (happens during a Magento upgrade)

### Expected result (*)

1. integration tests run

### Actual result (*)

1. Integration tests fail with error like `Call to undefined method Magento\Checkout\Api\GuestShippingInformationManagementTest::getAnnotations()`

Current composer.lock installs phpunit 9.2.6. With phpunit 9.5.0 we have this removal
```
-    /**
-     * @internal This method is not covered by the backward compatibility promise for PHPUnit
-     */
-    public function getAnnotations(): array
-    {
-        return TestUtil::parseTestMethodAnnotations(
-            \get_class($this),
-            $this->name
-        );
-    }
```
source https://github.com/sebastianbergmann/phpunit/compare/9.2.6...9.5.0.diff

which is used actively in integration tests.

```
user@3e1120e3426a:/var/www$ grep -r '>getAnnotations()' dev/tests/
dev/tests/integration/framework/Magento/TestFramework/Annotation/AdminConfigFixture.php:        $annotations = $test->getAnnotations();
dev/tests/integration/framework/Magento/TestFramework/Annotation/ComponentRegistrarFixture.php:        $annotations = $test->getAnnotations();
dev/tests/integration/framework/Magento/TestFramework/Annotation/Cache.php:        $source = $test->getAnnotations();
dev/tests/integration/framework/Magento/TestFramework/Annotation/IndexerDimensionMode.php:        $source = $test->getAnnotations();
dev/tests/integration/framework/Magento/TestFramework/Annotation/AppIsolation.php:        $annotations = $test->getAnnotations();
dev/tests/integration/framework/Magento/TestFramework/Annotation/ConfigFixture.php:        $annotations = $test->getAnnotations();
dev/tests/integration/framework/Magento/TestFramework/Annotation/AbstractDataFixture.php:        $annotations = $scope === null ? $this->getAnnotations($test) : $test->getAnnotations()[$scope];
dev/tests/integration/framework/Magento/TestFramework/Annotation/AbstractDataFixture.php:        $annotations = $test->getAnnotations();
dev/tests/integration/framework/Magento/TestFramework/Annotation/AppArea.php:        $area = $this->_getTestAppArea($test->getAnnotations());
dev/tests/integration/framework/Magento/TestFramework/Annotation/DbIsolation.php:        $annotations = $test->getAnnotations();
dev/tests/integration/framework/tests/unit/testsuite/Magento/Test/Annotation/AdminConfigFixtureTest.php:            ->willReturn($this->getAnnotations()['method'][$this->object::ANNOTATION]);
dev/tests/integration/framework/tests/unit/testsuite/Magento/Test/Annotation/ConfigFixtureTest.php:            ->willReturn($this->getAnnotations()['method'][$this->object::ANNOTATION]);
dev/tests/setup-integration/framework/Magento/TestFramework/Annotation/CopyModules.php:        $annotations = $test->getAnnotations();
dev/tests/setup-integration/framework/Magento/TestFramework/Annotation/CopyModules.php:        $annotations = $test->getAnnotations();
dev/tests/setup-integration/framework/Magento/TestFramework/Annotation/SchemaFixture.php:            $annotations = $test->getAnnotations()[$scope];
dev/tests/setup-integration/framework/Magento/TestFramework/Annotation/SchemaFixture.php:        $annotations = $test->getAnnotations();
dev/tests/setup-integration/framework/Magento/TestFramework/Annotation/DataProviderFromFile.php:        $annotations = $test->getAnnotations();
```

Temporary fix is to lock phpunit to lower than 9.5.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by fooman** [comment]
`composer require phpunit/phpunit:'^9.0 <9.5.0'`

**Comment by andrewbess** [comment]
This issue is closed as duplicate magento/magento2#33596
